app/src/main/java/ch/epfl/sdp/artificial_intelligence/RectangleBounds.java

```diff
@@ -1,36 +1,37 @@
 package ch.epfl.sdp.artificial_intelligence;
 
-import ch.epfl.sdp.GeoPoint;
+import ch.epfl.sdp.map.GeoPoint;
 
 public class RectangleBounds implements Boundable {
-    private float height;
-    private float width;
+    private float halfHeight;
+    private float halfWidth;
     private GeoPoint lowerLeft;
 
     public RectangleBounds(float height, float width, GeoPoint lowerLeft){
-        this.height = height;
-        this.width = width;
+        this.halfHeight = height / 2;
+        this.halfWidth = width / 2;
```

Comment:
Just wondering why we need halfHeight/halfWidth here? :)

Reply:
If you look at how the rectangle is computed: one side is a half and the other side is the other half. Also, I prefer to store it that way, because otherwise I have to recompute the divisions each time I call isInside().